### ISSUE TITLE
Thinkware dashcam complains about the flash memory card format

### ISSUE THREAD

**Issue by 130s** [kinu-garage/hut_10sqft#948]
Tech investigation for https://github.com/kinu-garage/30y-130s_life/issues/1861

**Comment by 130s** [comment]
Hmm...Plugging the SD card to Mac's external SD reader doesn't seem to automatically mount.

**Comment by 130s** [comment]
> Hmm...Plugging the SD card to Mac's external SD reader doesn't seem to automatically mount.

NVM. Replugged then it's now mounted.
- I do see many videos are recorded.
- At a glance don't notice any video missing. I do notice a video recording is missing for a few minutes even a car already started driving.
   - E.g. 2023/12/10/105403, which is 1 or 2 minutes after the car started. I vaguely remember the dashcam complained and said it'd restart.
- I see it formated as FAT32:
   ```
   % diskutil info -all

   Device Identifier:         disk2s1
   Device Node:               /dev/disk2s1
   Whole:                     No
   Part of Whole:             disk2

   Volume Name:               NO NAME
   Mounted:                   Yes
   Mount Point:               /Volumes/NO NAME

   Partition Type:            Windows_FAT_32
   File System Personality:   MS-DOS FAT32
   Type (Bundle):             msdos
   Name (User Visible):       MS-DOS (FAT32)

   OS Can Be Installed:       No
   Media Type:                Generic
   Protocol:                  USB
   SMART Status:              Not Supported
   Volume UUID:               ED4D3332-4C97-365D-9640-252A54D46488
   Partition Offset:          4194304 Bytes (8192 512-Byte-Device-Blocks)

   Disk Size:                 31.9 GB (31910789120 Bytes) (exactly 62325760 512-Byte-Units)
   Device Block Size:         512 Bytes

   Volume Total Space:        31.9 GB (31902400512 Bytes) (exactly 62309376 512-Byte-Units)
   Volume Used Space:         27.3 GB (27250786304 Bytes) (exactly 53224192 512-Byte-Units) (85.4%)
   Volume Free Space:         4.7 GB (4651614208 Bytes) (exactly 9085184 512-Byte-Units) (14.6%)
   Allocation Block Size:     32768 Bytes

   Media OS Use Only:         No
   Media Read-Only:           No
   Volume Read-Only:          No

   Device Location:           External
   Removable Media:           Removable
   Media Removal:             Software-Activated

   Solid State:               Info not available
   ```

**Comment by 130s** [comment]
TLDR; "Erase" is done on DiskUtility on Mac. FAT32 chosen. Will see if this resolve the issue.

https://www.thinkware.com/Support/Faq?cat=ACCESSORIES

> Q. How can I format the SD card?
> :
> 5. Select “ExFAT” for memory card sizes 64GB and above.
> Select “FAT32” for memory card sizes 32GB and lower.
> Click “Erase”, and then select “Done” once completed.

So I do see the current SD card is already FAT32, so whether it's really format issue or not is questionable..?

> Q. How can I diagnose for faulty SD card?
> :
> 6. If the following problem persists, format the SD card.

<details><summary>Ran Fast Aid on DiskUtility on Mac. Exit code 0.</summary>

```
Running First Aid on “NO NAME” (disk2s1)

Repairing file system.
Volume was successfully unmounted.
Performing fsck_msdos -y /dev/rdisk2s1
** /dev/rdisk2s1

** Phase 1 - Preparing FAT

** Phase 2 - Checking Directories

** Phase 3 - Checking for Orphan Clusters

649 files, 4542528 KiB free (141954 clusters)

File system check exit code is 0.
Restoring the original state found as mounted.

Operation successful.
```
</details>

Anyways, formatting now on DiskUtility on Mac. FAT32 chosen. Will see.

> Q. How long is the lifespan of a SD card?
> Please note that SD cards are consumable items. Due to the stress and build-up of bad blocks (spaces on the memory card that files can no longer be stored) due to constant overwriting cycles, SD cards typically have lifetime of 6 months to 1 year with regular formatting.
> When corrupt memory cards are inserted to Thinkware Dash Cams, our dash cam continually restarts in attempt to allocate space on the card to save its files.
> The dash cam may also state "Format SD card regularly for data integrity" upon each startup. Another indication that the SD card has gone corrupt is hearing the device say "Memory card error."
> If you believe that the SD card is beginning to become corrupt, insert the card into your PC. If the SD card is unable to be properly formatted it is time to replace this SD card. 

Really..? That's so short!

**Comment by 130s** [comment]
Thinkware F800 dashcam kept complaining that "memory card format is needed" and kept rebooting. I found pressing the "Format" button on the device' body immediately stops that and heard normal operation's announcement :/ :)

**Comment by 130s** [comment]
This issue keeps coming back and even the proposed workaround https://github.com/kinu-garage/hut_10sqft/issues/948#issuecomment-1875329915 doesn't always work. Recently I found I have to wait for several minutes, kept pressing reformat button before finally the device stops complaining.
I found out my wife simply unplugged the dashcam as she got annoyed. 

Trying out a different SD card.

**Comment by 130s** [comment]
Using a `gnome-disks` GUI, I just reformatted an SD card (that was used for some raspi purpose) to `FAT` (not `FAT32` as GUI didn't offer it), following https://support.thinkware.com/hc/en-us/articles/360045671873--Mac-How-to-Format-the-SD-Card
